test/failures_testsuite/testcases/s3_redundant/test04_minio_active_passive.py

```diff
@@ -1,23 +1,82 @@
+                    cls.config['s3_redundant']['instance']['shards'], cls.config['s3_redundant']['instance']['parity']]
+
+            for _ in range(5):
+                cls.s3_redundant_service_name = 's3_redundant_{}'.format(str(time.time()).split('.')[0])
+                cls.logger.info("s3 redundant service name : {}".format(cls.s3_redundant_service_name))
+
+                instance = cls.s3_redundant_controller.deploy_s3_redundant(cls.s3_redundant_service_name, *data)
+                try:
+                    cls.logger.info("wait for deploying {} service".format(cls.s3_redundant_service_name))
+                    instance.wait(die=True)
+                    break
+                except Exception as e:
+                    cls.logger.error("There is an error while installing s3 .. we will re-install it!")
+                    cls.logger.error(e)
+                    cls.logger.info('uninstall {} service'.format(cls.s3_redundant_service_name))
+                    s3_redundant_object = cls.s3_redundant_controller.s3_redundant[cls.s3_redundant_service_name]
+                    s3_redundant_object.uninstall()
+                    cls.logger.info('delete {} service'.format(cls.s3_redundant_service_name))
+                    s3_redundant_object.delete()
+            else:
+                raise TimeoutError("can't install s3 redundant .. gonna quit!")
+
+            cls.logger.info('wait for {} state to be okay'.format(cls.s3_redundant_service_name))
+            for _ in range(10):
+                cls.s3_redundant_object = cls.s3_redundant_controller.s3_redundant[cls.s3_redundant_service_name]
+                state = cls.s3_redundant_object.service.state
+                cls.logger.info("{} state : {}".format(cls.s3_redundant_service_name, state))
+                try:
+                    cls.logger.info("waiting {} state to be ok ... ".format(cls.s3_redundant_service_name))
+                    state.check('actions', 'install', 'ok')
+                    break
+                except:
+                    time.sleep(5 * 60)
+                    cls.logger.info("wait for 5 mins .. then we try again!")
+            else:
+                state.check('actions', 'install', 'ok')
+            cls.logger.info('wait for 120 sec before running test cases')
+            time.sleep(120)
+        else:
+            sub = Popen('zrobot godtoken get', stdout=PIPE, stderr=PIPE, shell=True)
+            out, err = sub.communicate()
+            god_token = str(out).split(' ')[2]
+            cls.s3_redundant_controller = Controller(cls.config, god_token)
+            cls.s3_redundant_service_name = cls.config['s3_redundant']['use']['s3_redundant_service_name']
+            cls.s3_redundant_object = cls.s3_redundant_controller.s3_redundant[cls.s3_redundant_service_name]
+
+        cls.s3_active_service_name = cls.s3_redundant_object.active_s3
```

Comment:
Also in case s3_red has been installed using the setup class .. i don't think s3_active and the s3_passive will inherit the properties of the s3.py which happens when u call ```Controller()
``` so i think you need to call init of Controller() after installation is done.